internal/services/tenant_isolation_policy/api_tenant_isolation_policy.go

```diff
@@ -0,0 +1,190 @@
+		finalPolicy, err := client.getTenantIsolationPolicy(ctx, tenantId)
+		if err != nil {
+			return nil, fmt.Errorf("Error getting tenant isolation policy after async operation: %v", err)
+		}
+
+		return finalPolicy, nil
+	}
+
+	return &updatedPolicy, nil
+}
+
+// getRetryAfterDuration parses the Retry-After header from an HTTP response.
+// It converts the header value to a time.Duration, defaulting to 5 seconds
+// if the header is missing or invalid.
+func getRetryAfterDuration(resp *http.Response) time.Duration {
+	// Default value if header not present or invalid
+	defaultDuration := 5 * time.Second
+
+	if resp == nil {
+		return defaultDuration
+	}
+
+	// Check for Retry-After header
+	retryAfter := resp.Header.Get("Retry-After")
+	if retryAfter == "" {
+		return defaultDuration
+	}
+
+	// Try to parse as seconds (integer)
+	seconds, err := strconv.Atoi(retryAfter)
+	if err == nil && seconds > 0 {
+		return time.Duration(seconds) * time.Second
+	}
+
+	// If header value is not a valid integer, return default
+	return defaultDuration
+}
+
+// doWaitForLifecycleOperationStatus polls an asynchronous operation until completion.
+// It follows the location header from the initial response, respects Retry-After
+// headers to control polling frequency, and implements exponential backoff with
+// minimum 2s and maximum 60s between attempts.
+//
+// Parameters:
+//   - ctx: Context for the request with cancellation and timeout capabilities
+//   - response: The initial API response containing operation location headers
+//
+// Returns:
+//   - *TenantIsolationPolicyDto: Always nil (caller should get fresh state)
+//   - error: Any error encountered during the polling process
+func (client *Client) doWaitForLifecycleOperationStatus(ctx context.Context, response *api.Response) (*TenantIsolationPolicyDto, error) {
+	locationHeader := response.GetHeader(constants.HEADER_LOCATION)
+	if locationHeader == "" {
+		locationHeader = response.GetHeader(constants.HEADER_OPERATION_LOCATION)
+	}
+	tflog.Debug(ctx, fmt.Sprintf("Location Header: %s", locationHeader))
+
+	if locationHeader == "" {
+		return nil, fmt.Errorf("no Location or Operation-Location header found in async response")
+	}
+
+	for {
+		apiResp, err := client.Api.Execute(ctx, nil, "GET", locationHeader, nil, nil, []int{http.StatusOK, http.StatusAccepted}, nil)
+		if err != nil {
+			return nil, fmt.Errorf("error polling operation status: %v", err)
+		}
+
+		tflog.Debug(ctx, fmt.Sprintf("Operation poll status: %s", apiResp.HttpResponse.Status))
+
+		// If we get a 200 OK, the operation is complete
+		if apiResp.HttpResponse.StatusCode == http.StatusOK {
+			// Return nil since we'll get the fresh state later
+			return nil, nil
+		}
+
+		// Get the next wait time from the Retry-After header if available
+		waitTime := getRetryAfterDuration(apiResp.HttpResponse)
+
+		// For safety, ensure we have a minimum wait time and cap the maximum
+		if waitTime < 2*time.Second {
```

Comment:
reuse exisiting retryafter time values

Reply:
this is just bounds checking what we get back, but I could see this being in the impelmentation of the function